packages/gaxios/src/teeny-request.ts

```diff
@@ -0,0 +1,278 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Agent, AgentOptions as HttpsAgentOptions} from 'https';
+import {AgentOptions as HttpAgentOptions} from 'http';
+import {PassThrough, Readable} from 'stream';
+import {TeenyStatistics} from './teeny-statistics.js';
+import {request, GaxiosOptions, GaxiosResponse} from './index.js';
+
+export interface CoreOptions {
+  method?: string;
+  timeout?: number;
+  gzip?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  json?: any;
+  headers?: Headers;
+  body?: string | {};
+  useQuerystring?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  qs?: any;
+  proxy?: string;
+  multipart?: RequestPart[];
+  forever?: boolean;
+  pool?: HttpsAgentOptions | HttpAgentOptions;
+}
+
+export interface OptionsWithUri extends CoreOptions {
+  uri: string;
+}
+
+export interface OptionsWithUrl extends CoreOptions {
+  url: string;
+}
+
+export type Options = OptionsWithUri | OptionsWithUrl;
+
+export interface Request extends PassThrough {
+  agent: Agent | false;
+  headers: Headers;
+  href?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface Response<T = any> {
+  statusCode: number;
+  headers: Headers;
+  body: T;
+  request: Request;
+  statusMessage?: string;
+}
+
+export interface RequestPart {
+  body: string | Readable;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface RequestCallback<T = any> {
+  (err: Error | null, response: Response, body?: T): void;
+}
+
+export class RequestError extends Error {
+  code?: number;
+}
+
+interface Headers {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [index: string]: any;
+}
+
+function teenyRequest(reqOpts: Options): Request;
+function teenyRequest(reqOpts: Options, callback: RequestCallback): void;
+function teenyRequest(
+  reqOpts: Options,
+  callback?: RequestCallback,
+): Request | void {
+  const opts = {...reqOpts};
+  const uri = ((opts as OptionsWithUri).uri ||
+    (opts as OptionsWithUrl).url) as string;
+
+  if (!uri) {
+    throw new Error('Missing uri or url in reqOpts.');
+  }
+
+  const gaxiosOptions: GaxiosOptions = {
+    url: uri,
+    method: opts.method,
+    timeout: opts.timeout,
+    headers: opts.headers as any, // Cast to any to avoid Headers type mismatch
+    responseType: callback ? 'json' : 'stream',
+    validateStatus: () => true, // teeny-request doesn't throw on status codes by default in the same way, or does it?
+    // teeny-request treats non-2xx as success but might process body.
+    // Actually teeny-request relies on node-fetch which doesn't throw.
+    // gaxios defaults to validateStatus: status >= 200 && status < 300.
+    // We should disable validation to match teeny-request behavior (it passes response to callback).
```

Comment:
These comments read more like a conversation than actual information. Could you please rephrase them and turn them into an assertion instead?